.github/workflows/test.yml

```diff
@@ -0,0 +1,47 @@
+name: tests
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.20.x
+          cache: true
+      - name: Run Linter
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.51.2
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.20.x
+          cache: true
+      - name: Build
+        run: go build -v ./...
+      - name: Run Tests
+        run: go test -json ./... > test-results.json
+      - name: Annotate Tests
+        if: always()
```

Comment:
I noticed you added this. I'd left it out even though it's in the official docs because I didn't know what it was for but I guess you discovered why! 😅 Just to confirm, I assumed that this is so that it always runs even though the previous step fails, right?

Reply:
>Just to confirm, I assumed that this is so that it always runs even though the previous step fails, right?

Correct. You'd never know _why_ your tests failed, only _that_ they failed. I guess you made sure tests always passed and never ran into the issue. 💪🏼

Reply:
Indeed they did in my case ... and I don't like adding configuration I don't understand. Thanks. TIL.